### PR TITLE
Clarify community creation form

### PR DESCRIPTION
**Pull request by alisapa**
A solution for: #985 

The new strings "display_name", "name_explain" and "display_name_explain" are intended to be "Display Name", "Name – used as the identifier for the community, cannot be changed." and "Display name — shown as the title on the community's page, can be changed." in English.

### CODE REVIEW

File: ui/src/components/community-form.tsx [hunk 1]
```diff
@@ -116,7 +116,18 @@ export class CommunityForm extends Component<
...
+                  class="btn btn-sm text-muted"
+                  data-tippy-content={i18n.t('name_explain')}
+                  onclick={linkEvent(this, this.showNameExplain)}
+                >
+                  <svg class="icon icon-inline">
+                    <use xlinkHref="#icon-help-circle"></use>
+                  </svg>
+                </a>
               </label>
+              <div id="name_explain_div">
```
Comment (dessalines) [inline ui/src/components/community-form.tsx:129]: Why are you not using templating here? Don't use ids.

File: ui/src/components/community-form.tsx [hunk 2]
```diff
@@ -270,6 +292,26 @@ export class CommunityForm extends Component<
...
+
+  showDisplayNameExplain() {
+    let element = document.getElementById("displayname_explain_div");
+    if (element.innerHTML == "") {
+      element.innerHTML = '<p class="ml-4 mb-2">' + i18n.t('display_name_explain') + '</p>';
+    }
+    else {
+      element.innerHTML = "";
+    }
+  }
```
Comment (dessalines) [inline ui/src/components/community-form.tsx:313]: Bad bad bad, never inject html like this in react. Make the template in the tsx, and hide it using a boolean:

```
showDisplayExplanation = !showDisplayExplanation;

...

{this.showDisplayExplanation && (
<div ...
)}
```

But also, don't do that above. Just have the `name_explanation` in the tippy, it doesn't need a whole div explanation.

**Comment by dessalines** [comment]
You also didn't commit your `en.json`. But mainly, this doesn't need a whole other div explanation... the info icon, and a tippy should be enough.

**Comment by alisapa** [comment]
Changed it to use templating. Committed the en.ts file now (I previously thought I was not supposed to do that because the translations folder, where it's located, is listed in .gitignore). The problem with only the tippy is that you can't hover on mobile, so there's no obvious way for a mobile user to see that explanation if it doesn't show it on click.

**Comment by dessalines** [comment]
> Changed it to use templating. Committed the en.ts file now (I previously thought I was not supposed to do that because the translations folder, where it's located, is listed in .gitignore).

Remove that file, its correctly in the .gitignore. The correct file is `en.json`, not `en.ts`.

> The problem with only the tippy is that you can't hover on mobile, so there's no obvious way for a mobile user to see that explanation if it doesn't show it on click.

You can click and hold, this is how it works for all current tippys on mobile. The div isn't necessary here, just the tippy.

File: ui/src/translations/en.ts [hunk 3]
```diff
@@ -0,0 +1,287 @@
+export const en = {
```
Comment (dessalines) [inline ui/src/translations/en.ts:1]: This file needs to be removed.

**Comment by dessalines** [comment]
Looks good, thanks! I'll make a few changes to make this better on mobile, like the other examples in the code.

**Pull request merged**